palette/src/luma/luma.rs

```diff
@@ -725,10 +726,44 @@ where
     }
 }
 
+impl<S, T> RelativeContrast for Luma<S, T>
+where
+    T: FloatComponent,
+    S: LumaStandard,
```

Comment:
An alternative could be to force this to be linear. But I'm not sure that's the way to go. Maybe it's better to go the opposite way this time and implement this trait for anything that can be converted to linear `Luma`. I think a blanket implementation would be possible.